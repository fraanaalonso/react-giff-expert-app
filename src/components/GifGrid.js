import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GiffGridItem';


export const GifGrid = ( {category} ) => {

    const {data: images, loading} = useFetchGifs(category);


    return (
        <>
        <h3>{ category }</h3>

        { loading && <p className="animate__animated animate__flash">Loading...</p> }
        <div className='card-grid'>
                {
                    //I unstructured the array images
                    images.map( (img) => (
                    <GifGridItem
                        key = { img.id }
                        {...img}
                        
                    />
                    ))
                }
        
            </div> 
         </>
    )
}
