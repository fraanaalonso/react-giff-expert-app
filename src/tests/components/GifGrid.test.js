import { shallow } from 'enzyme'
import React from 'react'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test GifGrid', () => {
    const category = 'One Punch';
    test('Must show GifGrid Snapshot', () => {

        const wrapper = shallow(<GifGrid category={ category }/>);
        expect( wrapper ).toMatchSnapshot();
    })

    test('Must show items when getting images', () => {
        
        //lets simulate that we are getting the information provided by our customHook
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cosa.png',
            title: 'Una cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={ category }/>);
        
        expect( wrapper ).toMatchSnapshot()



    })
    
    
})