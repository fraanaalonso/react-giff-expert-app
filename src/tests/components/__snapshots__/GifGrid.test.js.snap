// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid Must show GifGrid Snapshot 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test GifGrid Must show items when getting images 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
